### PR TITLE
moves to root

### PR DESCRIPTION
**Pull request by aricahunter**
git hack now changes directory to the git root directory before running steps.

### CODE REVIEW

**Comment by kevgo** [comment]
Super easy! Please add an integration test that starts out in a folder that doesn't exist on the main branch. Here are the respective specs for that situation when using `git sync`: https://github.com/Originate/git-town/tree/master/features/git-sync/folder_does_not_exist_on_main_branch

File: features/git-hack/folder_does_not_exist/branch_exists_locally.feature [hunk 1]
```diff
@@ -0,0 +1,38 @@
+Feature: git hack: starting a new feature from the main branch (with remote repo)
```
Comment (charlierudolph) [inline features/git-hack/folder_does_not_exist/branch_exists_locally.feature]: This feature name needs to be updated.
This file should probably live at
```
features/git-hack/on_feature_branch/folder_does_exist_on_main_branch.feature
```

File: features/git-hack/folder_does_not_exist/branch_exists_locally.feature [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+  As a developer working on a new feature on the main branch
+  I want to be able to create a new up-to-date feature branch and continue my work there
+  So that my work can exist on its own branch, code reviews remain effective, and my team productive.
```
Comment (charlierudolph) [inline features/git-hack/folder_does_not_exist/branch_exists_locally.feature]: This should be updated to something like https://github.com/Originate/git-town/blob/master/features/git-sync/folder_does_not_exist_on_main_branch/no_conflict.feature#L3-L5

**Comment by charlierudolph** [comment]
Feature test looks awesome. Thanks @aricahunter.

**Comment by kevgo** [comment]
@aricahunter is off the project, so I'm taking this over. I have added more feature tests, and wasted another 2 hours on Git's really strange side effects on user shells when run in subshells. From my perspective this edge case is small enough to not invest more energy into this.

**Comment by charlierudolph** [comment]
This stumped me for a bit but I believe the error that occurs when you run the command makes sense. 

Adding `-b` when running cucumber gives you a much better error since it will print the full backtrace. The only error I got without `-b` was `No such file or directory - getcwd (Errno::ENOENT)` with no stack trace.

So the `feature` has a `new_folder`, but its committed. Thus when you run `git-hack`, `new_folder` is not stashed because it is not uncommitted. Thus when you try to move into `new_folder` on the  `new-feature`, you get an ENOENT.

**Comment by kevgo** [comment]
Cool trick to use `-b`, this is pure gold! With your hint, I was able to make the test run by extending the GT framework to only cd into the initial directory if it exists. The problem here was that when we compile the list of steps, the directory exists, but later when the steps run it doesn't. I also added another spec where the new directory is not committed, so we cover the cases when the new directory is committed and when it is not, and `git hack` works beautifully in both now. Thanks for the help with this!

File: src/helpers/folder_helpers.sh [hunk 3]
```diff
@@ -6,7 +6,10 @@
 # Changes into the given directory
 function change_directory {
   local directory=$1
-  run_command "cd $directory"
+  local omit_if_doesnt_exist=$2
+  if [ -d "$directory" ] || [ -z "$omit_if_doesnt_exist" ]; then
+    run_command "cd $directory"
+  fi
```
Comment (charlierudolph) [inline src/helpers/folder_helpers.sh]: Thoughts on just creating a new function for this? `change_directory_if_exists`
That function wraps the checking and then just calls `change_directory`

File: features/git-hack/on_feature_branch/in_uncommitted_subfolder.feature [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+  I want to be able to extract my open changes into a new feature branch
+  So that I can get them reviewed separately from the changes on this branch.
+
+
+  Background:
+    Given I have a feature branch named "feature"
+    And the following commit exists in my repository
+      | BRANCH  | LOCATION         | MESSAGE       | FILE NAME    |
+      | main    | local and remote | main commit   | main_file    |
+      | feature | local and remote | folder commit | feature_file |
```
Comment (charlierudolph) [inline features/git-hack/on_feature_branch/in_uncommitted_subfolder.feature]: I don't think this commit is necessary

Comment (kevgo) [inline features/git-hack/on_feature_branch/in_uncommitted_subfolder.feature]: True. Removed.

File: src/helpers/folder_helpers.sh (hunk 3 above)
Comment (kevgo) [inline src/helpers/folder_helpers.sh]: Good idea.

**Comment by kevgo** [comment]
Committed in https://github.com/Originate/git-town/commit/94e5114e7acf695cb57f89e193bb34ddaaf1ae47
